src/xlsx.js

```diff
@@ -52,14 +53,16 @@ function extract(sheet, {range, headers = false} = {}) {
 function valueOf(cell) {
   if (!cell) return;
   const {value} = cell;
-  if (value && value instanceof Date) return value;
-  if (value && typeof value === "object") {
-    if (value.formula || value.sharedFormula)
+  if (value && typeof value === "object" && !(value instanceof Date)) {
+    if (value.formula || value.sharedFormula) {
       return value.result && value.result.error ? NaN : value.result;
-    if (value.richText) return value.richText.map((d) => d.text).join("");
+    }
+    if (value.richText) {
+      return richText(value);
+    }
     if (value.text) {
       let {text} = value;
-      if (text.richText) text = text.richText.map((d) => d.text).join("");
+      if (text.richText) text = richText(text);
```

Comment:
When would you have a value = {text: {richText: […]}} instead of {richText: […]}?

Reply:
I found this example in a spreadsheet from Ramona. I didn't really question all of the different richText permutations at that point, but this implies we should try to find more examples to make sure we cover all of them. I'd consider any new ones we find bugs to fix though.

Reply:
Maybe hyperlinked richText is {hyperlink: …, text: {richText: […]}} (instead of {hyperlink: …, richText: […]})?

Reply:
That was the specific Ramona case, but there was also no good reason the text had to be rich text. Her example also came from Numbers → Excel.